edgelet/edgelet-kube/src/convert/named_secret.rs

```diff
@@ -25,10 +25,10 @@ impl NamedSecret {
     }
 }
 
-impl<'a> TryFrom<(&str, ImagePullSecret<'a>)> for NamedSecret {
+impl TryFrom<(String, ImagePullSecret)> for NamedSecret {
```

Comment:
What's the reason behind moving from a &str to String?

Reply:
In the first commit, I pay a technical debt to not following rustlang guidelines. `From<T>`/`TryFrom<T>` implementation should consume `T` and "deconstruct" object. This way we delegate a caller a decision to get owned version of a `T` instead of making cloning parts of implementation of `T` inside `From<T>`/`TryFrom<T>`